docs/tutorials/writing_a_test.rst

```diff
@@ -0,0 +1,414 @@
+add to it:
+
+.. code-block:: yaml
+
+    # This is the name of your test. The full name of this test would be
+    # 'supermagic.basic'.
+    basic:
+
+        # This will display as the test summary when you run 'pav show tests'
+        summary: A basic supermagic run.
+
+        # This section defines how the test is built, mainly by detailing how
+        # to write a 'build.sh' script.
+        build:
+            # Pavilion will auto-extract this archive. The extracted directory
+            # will be your build directory.
+            source_location: supermagic-master.zip
+
+            # Each of these commands is added as a separate line to the
+            # build script.
+            cmds:
+                - gcc -o supermagic supermagic.c
+
+        # The run section defines how the create the 'run.sh' script.
+        run:
+            cmds:
+                # Each of these commands will be inserted into our run script.
+                - ./supermagic
+
+Use the command '``pav show tests``' to get a list of all known tests, including
+yours.
+
+Note:
+  The above config won't work, but that's intentional. We'll fix it over the
+  course of this tutorial.
+
+.. code-block:: shell
+
+    $ pav show tests
+
+    -----------------+------------------------
+     Name            | Summary
+    -----------------+------------------------
+    supermagic.basic | A basic supermagic run.
+
+If your suite or test is highlighted in red, there was an error in your
+config. Use '``pav show tests --err``' to get information on what the problem
+is in your yaml file.
+
+
+Test Building
+~~~~~~~~~~~~~
+
+The combined cryptographic hashes of the build source and build script will
+be the build name in <working_dir>/builds.
+
+For instance, if our build hash is 'ac3251801d831', we'll end up with a
+build directory like this:
+
+.. code-block:: text
+
+    <working_dir>/ac3251801d831/
+        Makefile.am
+        supermagic.c
+        supermagic.h
+        util/
+            ...
+        ...
+
+We'll also end up with a build script that looks like this:
+
+.. code-block:: bash
+
+    #!/bin/bash
+
+    # The first (and only) argument of the build script is the test id.
+    export TEST_ID=${1:-0}
+    export PAV_CONFIG_FILE=/home/bob/pav2/config/pavilion.yaml
+    source /home/bob/pav2/src/bin/pav-lib.bash
+
+    # Perform the sequence of test commands.
+    gcc -o supermagic supermagic.c
+
+When building the test Pavilion will run that script in the extracted build
+directory.
+
+Let's try it:
+
+.. code-block:: shell
+
+    $ pav run supermagic.basic
+
+    Test tutorial.basic run 72 building 787aceaa19ac9a21
+    Error building test:
+    status BUILD_FAILED - Build returned a non-zero result.
+    For more information, run 'pav log build 72'
+
+Oh no! Our build failed. Let's follow the suggestion, and look at the build
+log for our test. We can also use '``pav cat 72 build.sh``' to output the build
```

Comment:
Still want a spack-like `pav cd 72` to take you to the test_run directory of that test.

Reply:
Are you aware that it isn't possible?